<?xml version="1.0"?>

<launch>
  <arg name="scan_topic"     default="/scan"/>
  <arg name="odom_topic"     default="/igvc_robot/odom"/>
  <arg name="cmd_vel"        default="/igvc_robot/cmd_vel"/>
  <arg name="waypoints_file" default="$(find igvc2022)/config/gps_waypoints/qualification_waypoints.yaml"/>
  
  <!-- Node for LED contoroll -->
  <node pkg="igvc2022" type="LED_control.py" name="LED_controller" output="screen"/>
  
  <!-- The gmapping package provides laser-based SLAM (Simultaneous Localization and Mapping), 
       as a ROS node called slam_gmapping -->
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
    <remap from="scan" to="$(arg scan_topic)"/>
    <rosparam file="$(find igvc2022)/params/gmapping_common_params.yaml" command="load"/>
  </node>
  
  <!-- The move_base package provides an implementation of an action that, 
  given a goal in the world, will attempt to reach it with a mobile base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <remap from="/odom"    to="$(arg odom_topic)"/>
    <remap from="/cmd_vel" to="$(arg cmd_vel)"/>
    <rosparam file="$(find igvc2022)/params/costmap_common_params.yaml"          command="load" ns="global_costmap"/>
	 <rosparam file="$(find igvc2022)/params/foot_print.yaml"                     command="load" ns="global_costmap"/>
	 <rosparam file="$(find igvc2022)/params/costmap_common_params.yaml"          command="load" ns="local_costmap"/>
	 <rosparam file="$(find igvc2022)/params/foot_print.yaml"                     command="load" ns="local_costmap"/>
	 <rosparam file="$(find igvc2022)/params/local_costmap_params.yaml"           command="load"/>
	 <rosparam file="$(find igvc2022)/params/global_costmap_params.yaml"          command="load"/>
	 <rosparam file="$(find igvc2022)/params/move_base_params.yaml"               command="load"/>
	 <rosparam file="$(find igvc2022)/params/base_local_planner_params.yaml"      command="load"/>
	 <rosparam file="$(find igvc2022)/params/recovery_params.yaml"                command="load"/>
  </node>
  
  <!-- Node for rviz visualization -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find igvc2022)/rviz/x.rviz"/>
  
  
  <param name="waypoints_file" value="$(arg waypoints_file)"/>
  <!-- <node pkg="igvc2022" type="qualification_gps_goal_setter.py" name="qualification_gps_goal_setter" output="screen"/>-->
  
</launch>
